docs/payments/concepts/payment-rails.md

```diff
@@ -16,6 +16,62 @@ A Payment Rail is a smart contract-based payment channel that:
 4. **Provides security**: Locks funds to ensure payment obligations are met
 5. **Offers flexibility**: Allows for rate adjustments and termination conditions
 
+### Epoch Duration
+
+In the Filecoin network, one epoch is approximately **30 seconds** in clock time. This is a fundamental parameter in the system because:
+
+1. The `paymentRate` is specified as tokens per epoch
+2. The `lockupPeriod` is measured in epochs (e.g., 2880 epochs = 1 day)
+3. Settlement intervals and proof submission windows are measured in epochs
+
+When calculating costs and rates, this epoch duration must be considered:
+
+```javascript
+// Converting between time-based rates and epoch-based rates
+const dollarsPerMonth = 30.0;
+const secondsPerMonth = 30 * 24 * 60 * 60; // 30 days in seconds
+const secondsPerEpoch = 30; // 30 seconds per epoch
+const epochsPerMonth = secondsPerMonth / secondsPerEpoch; // ~86,400 epochs
+const dollarsPerEpoch = dollarsPerMonth / epochsPerMonth;
+```
+
+### Payment Rate Calculation
+
+The `paymentRate` parameter represents the rate at which funds flow from the client to the storage provider per epoch. This rate is typically calculated off-chain based on several factors:
+
+```javascript
+// Example of calculating payment rate based on data size and market factors
+function calculatePaymentRate(dataSize, storagePrice, epochsPerDay) {
+    // Convert data size to GB
+    const dataSizeGB = dataSize / (1024 * 1024 * 1024);
+    
+    // Calculate daily cost based on market price per GB
+    const dailyCost = dataSizeGB * storagePrice;
+    
+    // Convert to per-epoch rate
+    const paymentRate = ethers.utils.parseUnits(
+        (dailyCost / epochsPerDay).toFixed(6),
+        6
+    );
+    
+    return paymentRate;
+}
+
+// Usage example
+const dataSize = 5 * 1024 * 1024 * 1024; // 5 GB in bytes
+const storagePrice = 0.01; // $0.01 per GB per day
+const epochsPerDay = 2880; // 30-second epochs
+const rate = calculatePaymentRate(dataSize, storagePrice, epochsPerDay);
+```
+
+Factors that typically influence the payment rate include:
+
+1. **Data Size**: Larger data requires more storage resources
+2. **Storage Duration**: Longer commitments may have different pricing
+3. **Market Rates**: Competitive pricing based on current market conditions
+4. **Quality of Service**: Premium service levels may command higher rates
```

Comment:
What or where are the service levels?  Is this associated with a given payment rate?